packages/app-store/routing-forms/lib/handleResponse.ts

```diff
@@ -185,25 +188,28 @@ const _handleResponse = async ({
     } else {
       // It currently happens for a Router route. Such a route id isn't present in the form.routes
     }
-
     let dbFormResponse;
     if (!isPreview) {
-      dbFormResponse = await prisma.app_RoutingForms_FormResponse.create({
-        data: {
-          // TODO: Why do we not save formFillerId available in the input?
-          // formFillerId,
+      if (queueFormResponse) {
```

Comment:
We either write the form response to the queued form response table or record at moment of submission